evap/evaluation/models.py

```diff
@@ -1044,6 +1044,16 @@ def unlogged_fields(self):
     def is_general(self):
         return self.contributor_id is None
 
+    @property
+    def can_be_deleted(self):
```

Comment:
I'm a bit uneasy about the following scenario:
- Issue for a new requirement: Contributions should not be deletable anymore if X.
- User stumbles across this method, and just adds `if X: return False´, thinking that it solves the issue
- The delete button will not be shown anymore -- but malicious staff users will be able to delete the instances. Also: Do we have other locations where Contributions can be deleted?

Considering this, I'd say for maintainability, it would be best if we either added this as a proper permissions check to wherever a contribution is deleted, or renamed it to "show_delete_button" or something. The latter could contain an explaining comment.

Reply:
Makes sense. As far as I know this is the only place where contributions are explicitly deleted.
Currently, deleting contributions (with answers) is not possible for any user (including malicious staff users) becaase of the database protection. So when we care about deletion protection, this would only be useful if there are other conditions (like the case you described) in which the database would actually allow the deletion. That leads me to the point that this would be caring for future conditions, which is not necessarily bad, but I am not sure if it should be handled right now and here?
If we want to do it I would probably override the `delete` method so we dont have to search for all places where contributions are deleted.

Otherwise I would probably just rename it to "show_delete_button", as you mentioned. What bothers me about this is that the model implements view logic, which does not really fit, right?

What do you think?

Reply:
Totally agree. I'd suggest renaming the method and moving it into the form.

Reply:
Good Idea 👍